modules/common/endpoint/endpoint.go

```diff
@@ -184,8 +196,11 @@ func ExposeEndpoints(
 					}),
 					exportLabels,
 					timeout,
+					data.RouteOverride,
 				)
 
+				// TODO create TLS cert if non provided in override and data.Protocol == ProtocolHTTPS
+
```

Comment:
> We could allow the user to provide the annotations required for cert-manager since they already have access to routeOverrides via the other piece of work: #293
> 
> Maybe we should preference that if users want TLS rather than implementing our own logic to generate TLS certs?

sure, for public endpoints (route) a user can overwrite certs for public endpoints via my other  PR you mentioned. but for tls-e need to create them based on the cert-manager issuer . I don't think for those it will work well if a user has to provide all the certs.